### DISCUSSION TITLE
How to view documentation from previous version

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#17548] [Help] [answered]
As far as I can tell there's no version switcher in the documentation. I'm working with an app using Next version 7 and I want to keep changes minimal as a rewrite occurs in a different branch. What's the best way to go back and look at the documentation as it applies to version 7 of Next.js?

**Comment by lfades** [comment, marked as answer]
Versions older than 8.0.0 iirc had a single readme for docs, for Next.js 7 you can see the docs here: https://github.com/vercel/next.js/tree/7.0.0

**Reply by canda** [reply]
and what about newer releases?

**Reply by james-tractable** [reply]
I can't find any way to view version 10 docs for example, and its not clear to me if some of the things the current docs say I should be able to do apply to version 10 or not. Is there not anywhere the older docs are archived?

**Reply by Asgaroth** [reply]
This was indeed an issue for me. Specifically the docs about custom `tsconfig`  I was pulling my hair only to realize It's not supported in v10.

TBF it's quite easy to upgrade next to new versions but still, versioned docs would be nice to have.

**Reply by nir099** [reply]
Agreed on this, find v10 source code on https://github.com/vercel/next.js/tree/v10.2.0

**Reply by arvinsim** [reply]
@nir099 What command did you run after cloning to access the docs?

**Reply by canda** [reply]
you could navigate to the docs folder and look for your tag
https://github.com/vercel/next.js/tree/v13.3.3/docs
<img width="572" alt="image" src="https://github.com/vercel/next.js/assets/1220263/48215ed2-00ef-48b2-8315-da65daf370c6">

but since github does only indexes the default branch, you can't search through the github ui
you'll have to clone the repo and search locally
it's not very convinient

**Comment by rdebeasi** [comment]
It's also possible to specify a tag after `docs` in the URL to view docs as of that tag. For example: https://nextjs.org/docs/tag/v10.2.0/basic-features/data-fetching

**Reply by rahimimo** [reply]
It worked for me thanks. At first I thought you are proposing this, but actually this is working

**Reply by aaronkelton** [reply]
Link now returns a 404. Is there an updated way to do this? The latest equivalent page would be https://nextjs.org/docs/app/building-your-application/data-fetching/fetching

**Reply by rdebeasi** [reply]
Hm, interesting. That link no longer works for me either. Looks like the original suggestion to view the documentation files on GitHub might be a better option.

**Reply by cmaring** [reply]
The link with he tag no longer works. Is there any specific reason for disallowing us to view docs for a previous version?

**Reply by 144mdgross** [reply]
Same. This also 404s for me.
It seems like if this was previously supported with a URL, it would be achievable to add a selector to the UI to adjust the version as well. Having to clone the repo just to have reasonable search options is a disappointing reality for developers. It's common for many businesses to be running earlier versions of software and making that information hard to find feels like an oversight that makes working with nextJS needlessly annoying and slow. Please add versioned docs soon!

**Comment by NakZMichael** [comment]
It is not easy to view, but we can check it out on GitHub.
For example, https://github.com/vercel/next.js/blob/v12.3.4/docs/basic-features/image-optimization.md

**Comment by 4leite** [comment]
Please where are the v13 docs?
